provider/bidengine/order_test.go

```diff
@@ -0,0 +1,591 @@
+
+	order, err := newOrder(serviceCast, scaffold.orderId, bid, pricing)
+
+	require.Nil(t, err)
+	require.NotNil(t, order)
+
+	return order, scaffold
+}
+
+func Test_BidOrderAndUnreserve(t *testing.T) {
+	order, scaffold := makeOrderForTest(t, nil, nil)
+
+	firstBroadcast := <-scaffold.testClient.ftc.broadcasts
+	// Should have called reserve once
+	require.Equal(t, 1, scaffold.testCluster.reserveCallCount)
+	require.Equal(t, scaffold.orderId, scaffold.testCluster.lastReserveCall.orderId)
+
+	require.NotNil(t, firstBroadcast)
+	require.Len(t, firstBroadcast, 1)
+	broadcast := firstBroadcast[0]
+
+	createBidMsg, ok := broadcast.(*mtypes.MsgCreateBid)
+	require.True(t, ok)
+
+	require.Equal(t, createBidMsg.Order, scaffold.orderId)
+
+	priceDenom := createBidMsg.Price.Denom
+	require.Equal(t, "akt", priceDenom)
+	priceAmount := createBidMsg.Price.Amount.Int64()
+
+	require.GreaterOrEqual(t, priceAmount, int64(1))
+	require.Less(t, priceAmount, int64(100))
+
+	// After the broadcast call shut the thing down
+	// and then check what was broadcast
+	order.lc.Shutdown(nil)
+
+	// Should have called unreserve once, nothing happened after the bid
+	require.Equal(t, 1, scaffold.testCluster.unreserveCallCount)
+
+	require.Equal(t, scaffold.orderId, scaffold.testCluster.lastUnreserveCall.orderId)
+}
+
+func Test_BidOrderAndThenClosedUnreserve(t *testing.T) {
+	order, scaffold := makeOrderForTest(t, nil, nil)
+
+	<-scaffold.testClient.ftc.broadcasts
+	// Should have called reserve once at this point
+	require.Equal(t, 1, scaffold.testCluster.reserveCallCount)
+	require.Equal(t, scaffold.orderId, scaffold.testCluster.lastReserveCall.orderId)
+
+	ev := mtypes.EventOrderClosed{
+		Context: sdkutil.BaseModuleEvent{},
+		ID:      scaffold.orderId,
+	}
+	err := scaffold.testBus.Publish(ev)
+	require.Nil(t, err)
+
+	// Wait for this to complete. An order close event has happened so it stops
+	// on its own
+	<-order.lc.Done()
+
+	// Should have called unreserve once
+	require.Equal(t, 1, scaffold.testCluster.unreserveCallCount)
+
+	require.Equal(t, scaffold.orderId, scaffold.testCluster.lastUnreserveCall.orderId)
+}
+
+func Test_BidOrderAndThenLeaseCreated(t *testing.T) {
+	order, scaffold := makeOrderForTest(t, nil, nil)
+
+	// Wait for first broadcast
+	firstBroadcast := <-scaffold.testClient.ftc.broadcasts
+	require.NotNil(t, firstBroadcast)
+	require.Len(t, firstBroadcast, 1)
+	broadcast := firstBroadcast[0]
+
+	createBidMsg, ok := broadcast.(*mtypes.MsgCreateBid)
+	require.True(t, ok)
+	require.Equal(t, createBidMsg.Order, scaffold.orderId)
+	priceDenom := createBidMsg.Price.Denom
+	require.Equal(t, "akt", priceDenom)
+	priceAmount := createBidMsg.Price.Amount.Int64()
+
+	require.GreaterOrEqual(t, priceAmount, int64(1))
+	require.Less(t, priceAmount, int64(100))
+
+	leaseId := mtypes.LeaseID{
+		Owner:    "",
+		DSeq:     scaffold.groupId.DSeq,
+		GSeq:     scaffold.groupId.GSeq,
+		OSeq:     4,
+		Provider: scaffold.testAddr,
+	}
```

Comment:
do we have the order id here?  `MakeLeaseID` if not (please add if it's missing).